examples/using-js-search/src/components/ClientSearch.js

```diff
@@ -0,0 +1,220 @@
+import React, { Component } from 'react'
+import * as JsSearch from 'js-search'
+
+class ClientSearch extends Component {
+  state = {
+    isLoading: true,
+    searchResults: [],
+    search: null,
+    isError: false,
+    indexByTitle: false,
+    indexByAuthor: false,
+    termFrequency: true,
+    removeStopWords: false,
+    searchQuery: '',
+    selectedStrategy: '',
+    selectedSanitizer: '',
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (prevState.search === null) {
+      const { engine } = nextProps
+      return {
+        indexByTitle: engine.TitleIndex,
+        indexByAuthor: engine.AuthorIndex,
+        termFrequency: engine.SearchByTerm,
+        selectedSanitizer: engine.searchSanitizer,
+        selectedStrategy: engine.indexStrategy,
+      }
+    }
+    return null
+  }
+  async componentDidMount() {
+    this.rebuildIndex()
+  }
+
+  rebuildIndex = () => {
+    const {
+      selectedStrategy,
+      selectedSanitizer,
+      removeStopWords,
+      termFrequency,
+      indexByTitle,
+      indexByAuthor,
+    } = this.state
+    const { books } = this.props
+
+    const dataToSearch = new JsSearch.Search('isbn')
+    if (removeStopWords) {
+      dataToSearch.tokenizer = new JsSearch.StopWordsTokenizer(
+        dataToSearch.tokenizer
+      )
+    }
+    if (selectedStrategy === 'All') {
+      dataToSearch.indexStrategy = new JsSearch.AllSubstringsIndexStrategy()
+    }
+    if (selectedStrategy === 'Exact match') {
+      dataToSearch.indexStrategy = new JsSearch.ExactWordIndexStrategy()
+    }
+    if (selectedStrategy === 'Prefix match') {
+      dataToSearch.indexStrategy = new JsSearch.PrefixIndexStrategy()
+    }
+    selectedSanitizer === 'Case Sensitive'
+      ? (dataToSearch.sanitizer = new JsSearch.CaseSensitiveSanitizer())
+      : (dataToSearch.sanitizer = new JsSearch.LowerCaseSanitizer())
+    termFrequency === true
```

Comment:
eslint complains about this and the previous statement. You could restructure the statements to something like:

```js
    dataToSearch.searchIndex =
      termFrequency === true
        ? new JsSearch.TfIdfSearchIndex(`isbn`)
        : new JsSearch.UnorderedSearchIndex()
```

or just disable eslint for these lines with `// eslint-disable-next-line no-unused-expressions`

Reply:
the example files were not linted on purpose. I was aiming for a review on the documentation first, then lint the files and commit them